src/main/java/spark/route/RouteEntry.java

```diff
@@ -95,33 +97,28 @@ private boolean matchPath(String path) { // NOSONAR
         } else {
             // Number of "path parts" not the same
             // check wild card:
-            if (this.path.endsWith("*")) {
-                if (pathSize == (thisPathSize - 1) && (path.endsWith("/"))) {
-                    // Hack for making wildcards work with trailing slash
-                    pathList.add("");
-                    pathList.add("");
-                    pathSize += 2;
-                }
+            if (this.path.contains("*") && thisPathSize < pathSize) {
```

Comment:
would it be better to use string.indexof('*') so its a char match?

Reply:
this looks more obvious in comparison to `this.path.indexOf('*') >= 0 && thisPathSize < pathSize` hence kept it this way.